### DISCUSSION TITLE
Routing queries with flask

### DISCUSSION THREAD

**Discussion by ayushkhanduri** [pallets/flask#4199] [Q&A]
Hi , I am currently working with flask and I am confused about how the routing works in this microframework. I have collected a few questions.

Can you use @app.route on Class Methods instead of just functions?

can I write custom views using the MethodView approach?

eg for an user entity , ‘user/findUserByEmail’.

**Comment by davidism** [comment]
> Can you use @app.route on Class Methods instead of just functions?

No, it doesn't make sense to decorate class methods with `route`.

> can I write custom views using the MethodView approach?

I'm not sure what you mean. Yes, you can write whatever views you want. Using `MethodView` is one way to do that.

**Reply by ayushkhanduri** [reply]
what I mean was , method view provides interface for implementing HTTP Methods, let's suppose I have an entity named User and I have a GET request which searches user by email and I want my name my route user/findUserByEmail/<email>. Can I do that in User class? 

Bottom Line: I have a custom route , other than the HTTP Method and I want to implement that in the same class

`class User(MethodView):

    def getUserByEmail(self):
        # logic
